benchmarks/common/benchmark_runner.hpp

```diff
@@ -293,86 +303,38 @@ struct BenchmarkRunner {
-      Base::initialize(problem_size);
-    }
+      std::stringstream full_test_name;
+      full_test_name << test_name << "/";
+      std::string test_name_suffix = std::to_string(options.m) + "x" + 
+                                     std::to_string(options.n) + "x" + 
+                                     std::to_string(options.k) + "x" + 
+                                     std::to_string(options.l);
+      full_test_name << test_name_suffix;
+      benchmark::RegisterBenchmark(full_test_name.str().c_str(), run_benchmark, options, gemm_op)
+      ->UseManualTime();
+      benchmark::RunSpecifiedBenchmarks();
+  }
+
+  ~BenchmarkRunner() {
+    benchmark::Shutdown();
+  }
     
-    void run(const Options& options, const cutlass::KernelHardwareInfo& hw_info) override {
-      ProblemShapeType problem_size = ProblemShapeType{options.m, options.n, options.k, options.l};
-
-      initialize(problem_size);
-
-      typename Gemm::GemmKernel::Arguments arguments{
-              cutlass::gemm::GemmUniversalMode::kGemm,
-              problem_size,
-              {Base::block_A.get(), Base::stride_A, Base::block_B.get(), Base::stride_B},
-              {
-                {options.alpha, options.beta},
-                Base::block_C.get(), Base::stride_C, Base::block_D.get(), Base::stride_D
-              },
-              hw_info
-      };
-
-      Gemm gemm_op;
-
-      size_t workspace_size = Gemm::get_workspace_size(arguments);
-      cutlass::device_memory::allocation<uint8_t> workspace(workspace_size);
-
-      gemm_op.can_implement(arguments);
-
-      gemm_op.initialize(arguments, workspace.get());
-
-      // Run the GEMM
-      gemm_op.run();
-
-#if defined(CUTLASS_ENABLE_SYCL)
-      syclcompat::wait();
-#else
-      cudaDeviceSynchronize();
-#endif
-
-      // Verify that the result is correct
-      bool passed = Base::verify(problem_size, options.alpha, options.beta);
-      std::cout << "Disposition: " << (passed ? "Passed" : "Failed") << std::endl;
-
-      if (passed && options.iterations > 0) {
+  private:
+    static void run_benchmark(benchmark::State& state, const Options& options,  Gemm gemm_op) {
+      for(auto _ : state) {
         GPU_Clock timer;
         timer.start();
-        for (int i = 0; i < options.iterations; ++i) {
-          gemm_op.run();
-        }
-
-        float cute_time = timer.seconds() / options.iterations;
-        double tflops = (2.0 * options.m * options.n * options.k * options.l) * 1e-12;
-        std::cout << "Problem Size: " << options.m << 'x' << options.n << 'x' << options.k << 'x' << options.l << std::endl;
-        printf("Cutlass GEMM Performance:     [%4.3f]TFlop/s  (%6.4f)ms\n", tflops / cute_time, cute_time*1000);
+        gemm_op.run(); 
+        float cute_time = timer.seconds();
+        state.counters["TFlops"] = ((2.0 * options.m * options.n * options.k * options.l) * 1e-12) / cute_time;
+        state.counters["Time Elapsed (ms)"] = cute_time * 1000;
```

Comment:
```suggestion
state.counters["runtime_ms"] += cute_time * 1000;
```

Same here: you can use `+=` to get the total runtime.
I changed to name to `runtime_ms` to make it easy to parse.